documentation/src/main/asciidoc/topics/con_eviction_overview.adoc

```diff
@@ -0,0 +1,87 @@
+[id='how_eviction_works-{context}']
+= How Eviction Works
+{brandname} eviction relies on two configurations:
+
+* Size of the data container.
+* Eviction strategy.
+
+.Calculating data container size
+
+You configure the maximum size of the data container and specify if {brandname}
+stores cache entries as:
+
+* Object in the Java heap.
+* Binary `byte[]` in the Java heap.
+* Binary `byte[]` in native memory (off-heap).
+
+|===
+| Storage type | Size of the data container is calculated as:
+
+| Object
+| Number of entries.
+
+| Binary
+| Number of entries, if the eviction type is `COUNT`. +
+Amount of memory, in bytes, if the eviction type is `MEMORY`.
+
+| Off-heap
+| Number of entries, if the eviction type is `COUNT`. +
+Amount of memory, in bytes, if the eviction type is `MEMORY`.
+|===
+
+To calculate the data container size, {brandname} does one of the following:
+
+* Iterates over caches to count the number of entries to determine size.
```

Comment:
It doesn't iterate, it just keeps a running counter as entries are inserted or removed.

I don't know if we need to mention this part tbh.